frontend/components/user/settings/UserSettingsPassword.vue

```diff
@@ -0,0 +1,147 @@
+<script lang="ts" setup>
+import { object as yupObject } from 'yup'
+import { useForm } from 'vee-validate'
+import { API_PATH } from '~/types/customFetch'
+
+const { t: $t } = useI18n()
+const { fetch } = useCustomFetch()
+const toast = useBcToast()
+
+const { handleSubmit, errors, defineField } = useForm({
+  validationSchema: yupObject({
+    oldPassword: passwordValidation($t),
+    newPassword: passwordValidation($t),
+    confirmPassword: confirmPasswordValidation($t, 'newPassword')
+  })
+})
+
+const [oldPassword, oldPasswordAttrs] = defineField('oldPassword')
+const [newPassword, newPasswordAttrs] = defineField('newPassword')
+const [confirmPassword, confirmPasswordAttrs] = defineField('confirmPassword')
+
+const buttonsDisabled = defineModel<boolean | undefined>({ required: true })
+
+const onSubmit = handleSubmit(async (values) => {
+  if (!canSubmit.value) {
+    return
+  }
+
+  buttonsDisabled.value = true
+  try {
+    await fetch(API_PATH.USER_CHANGE_PASSWORD, {
+      body: {
+        password: values.newPassword
+      }
+    })
```

Comment:
As for the email, please give the user a positive feedback on success

Reply:
Good idea, added. Furthermore the form now gets reset on success.

Reply:
same - A+